baseapp/baseapp.go

```diff
@@ -616,7 +618,10 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 				panic(sdk.ErrorGasOverflow{Descriptor: "tx gas summation"})
 			}
 		}
-	}()
+	}
+	if mode == runTxModeDeliver {
+		defer consumeBlockGas()
```

Comment:
This defer behavior is unchanged and still runs if we do not make it to runMessages for a transaction